components/list/list-header.js

```diff
@@ -93,14 +125,28 @@ class ListHeader extends RtlMixin(LocalizeCoreElement(LitElement)) {
 	constructor() {
 		super();
 		this.noSelection = false;
+		this.noSticky = false;
 		this.paddingType = 'normal';
 		this.selectAllPagesAllowed = false;
 		this.slim = false;
 	}
 
+	connectedCallback() {
+		super.connectedCallback();
+
+		const parent = findComposedAncestor(this.parentNode, node => node && node.tagName === 'D2L-LIST');
+		if (!parent) return;
+
+		this._extendSeparator = parent.hasAttribute('extend-separators');
```

Comment:
Suggest defaulting this to `false` in the constructor otherwise it'll be `undefined` if no parent is found.